libcontainer/cgroups/systemd/apply_systemd.go

```diff
@@ -450,7 +450,14 @@ func (m *Manager) Set(container *configs.Config) error {
 }
 
 func getUnitName(c *configs.Cgroup) string {
-	return fmt.Sprintf("%s-%s.scope", c.Parent, c.Name)
+	allowed := "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ:-_.\\"
+	sanitizeFunc := func(r rune) rune {
+		if strings.ContainsRune(allowed, r) {
+			return r
+		}
+		return '.'
+	}
+	return fmt.Sprintf("%s-%s.scope", strings.Map(sanitizeFunc, c.Parent), strings.Map(sanitizeFunc, c.Name))
```

Comment:
I'm not sure the replacement is a good idea, and as I said in #336 , this "parent-name.scope" is not a good idea in the first place, specially when we support assigning a slice as the parent, so we should change this, WDYT?